Plugins/SequencePlugin/Source/SequencePlugin/Private/SystemDataBuilder.cpp

```diff
@@ -302,33 +287,32 @@ void USystemDataBuilder::InitGetContactData()
 /*
 * We expect to receive an authable wallet, a proper chainId, and PublicAddress and a valid indexer
 */
-void USystemDataBuilder::InitBuildSystemData(USequenceWallet* Wallet, int64 ChainId, FString PublicAddress, ASequenceBackendManager* Manager)
+void USystemDataBuilder::InitBuildSystemData(USequenceWallet* Wallet, ASequenceBackendManager* Manager)
 {
 	this->GWallet = Wallet;
-	this->GChainId = ChainId;
-	this->GPublicAddress = PublicAddress;
 	this->SqncMngr = Manager;
 	this->MasterSyncer->OnDoneDelegate.BindUFunction(this, "OnDone");
 
 	//sync operations FIRST
 	//all of this we get from auth!
 	this->SystemDataGuard.Lock();
-	this->SystemData.user_data.public_address = PublicAddress;
+	this->SystemData.user_data.public_address = Wallet->GetWalletAddress();
 	this->SystemData.user_data.account_id = this->SqncMngr->GetUserDetails().AccountID;
 	this->SystemData.user_data.email = this->SqncMngr->GetUserDetails().Email;
 	this->SystemData.user_data.email_service = this->SqncMngr->GetUserDetails().EmailService;
 	this->SystemData.user_data.username = this->SqncMngr->GetUserDetails().Username;
 	FNetwork_BE default_network;
 	default_network.is_default = true;
-	default_network.network_name = USequenceSupport::GetNetworkName(this->GChainId);
+	default_network.network_name = USequenceSupport::GetNetworkName(Wallet->GetNetworkId());
 	this->SystemData.user_data.networks.Add(default_network);
 	this->SystemDataGuard.Unlock();
+	
 	//ASYNC Operations next!
-	this->MasterSyncer->Increase(4);//+1 for each General Operation you have here!
+	this->MasterSyncer->Increase(1); //+1 for each General Operation you have here!
 	this->InitGetQrCode();
-	this->InitGetTokenData();
-	this->InitGetTxnHistory();
-	this->InitGetContactData();
+	//this->InitGetTokenData();
+	//this->InitGetTxnHistory();
+	//this->InitGetContactData();
```

Comment:
I'd vote to just remove over commenting out

Reply:
I removed all async calls there, including QR, all those didn't work anyway. Also included the missing struct inside SequenceSendIntent.h